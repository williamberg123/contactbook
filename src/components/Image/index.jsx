import PropTypes from 'prop-types';
import StyledImage from './styles';

export default function Image({ source }) {
	return (
		<StyledImage src={source} />
	);
}

Image.propTypes = {
	source: PropTypes.string,
};
