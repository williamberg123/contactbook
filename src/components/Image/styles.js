import styled from 'styled-components';

const StyledImage = styled.img`

`;

export default StyledImage;
