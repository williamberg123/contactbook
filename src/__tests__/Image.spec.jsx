import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Image from '../components/Image';

describe('<Image />', () => {
	it('should render Image with the correctly source', () => {
		expect.assertions(2);
		const imageSource = 'img/img.png';

		render(<Image source={imageSource} />);
		expect(screen.getByRole('img')).toBeInTheDocument();
		expect(screen.getByRole('img')).toHaveAttribute('src', imageSource);
	});

	it('should match snapshot', () => {
		const imageSource = 'img/img.png';

		render(<Image source={imageSource} />);
		expect(screen.getByRole('img')).toMatchSnapshot();
	});
});
