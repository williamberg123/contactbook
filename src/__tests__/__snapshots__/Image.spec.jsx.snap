// Vitest Snapshot v1

exports[`<Image /> > should match snapshot 1`] = `
<img
  class=""
  src="img/img.png"
/>
`;
